src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1372,18 +1375,52 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
               // ancestors; dataflow recurs on children when parents
               // move (to support partial (re)inits).
               //
-              // (I.e. querying parents breaks scenario 8; but may want
+              // (I.e. querying parents breaks scenario 7; but may want
               // to do such a query based on partial-init feature-gate.)
         }
+    }
+
+    fn check_if_path_or_subpath_is_moved(
+        &mut self,
+        context: Context,
+        desired_action: InitializationRequiringAction,
+        place_span: (&Place<'tcx>, Span),
+        flow_state: &Flows<'cx, 'gcx, 'tcx>,
+    ) {
+        // FIXME: analogous code in check_loans first maps `place` to
```

Comment:
what code are you referring to exactly?

Reply:
Exactly same code `let place = self.base_path(place_span.0);` is in `check_if_full_path_is_moved` also with the comment. So If somebody decides to refactor this he will probably do both `check_if_full_path_is_moved`, `check_if_path_or_subpath_is_moved`